Refactors mapTiles example to render-on-change

Brakebein opened the pull request:
Changes the mapTiles example to use an event-driven, render-on-change approach. This improves performance by reducing unnecessary renders. Listens for control changes, tile loading events, and window resize events to trigger rendering.

Adds rxjs dependency as devDependency for event handling and throttling.

Closes #1082

gkjohnson commented:
Thanks for making a PR - do you mind adjusting it so it doesn't require any dependencies like rxjs? It should be easy enough to make a custom debounced render function:

```js
function debounce( callback ) {

  let scheduled = false;
  return () => {

    if ( ! scheduled ) {

      queueMicrotask( () => {
        
        called = false;
        callback();

      } );

    }

  };
}
```

Brakebein commented:
It needs to be a throttled render function instead of debounced one, otherwise we won't properly see any updates. I changed it now to vanilla js. It constantly renders as long there are controls or tiles updates, and stops rendering when there are no update events triggered anymore.

gkjohnson commented on example/mapTiles.js:
```diff
@@ -122,6 +125,17 @@ function initTiles() {
+	tiles.addEventListener( 'load-model', render$ );
+	tiles.addEventListener( 'force-rerender', render$ );
+
+	// workaround for initial loading
+	tiles.addEventListener( 'load-tile-set', render );
```
Why do you feel this is a workaround? There's no reason to not debounce this, as well.

gkjohnson commented:
> It needs to be a throttled render function instead of debounced one, otherwise we won't properly see any updates.

This is indicative of bug - specifically #1081. Debouncing should work just fine so I've adjusted the demo to use it and I'll review this PR once I've fixed the #1081.

gkjohnson commented on package-lock.json:
```diff
@@ -5597,6 +5597,19 @@
         "node": ">=10.0.0"
       }
     },
+    "node_modules/concurrently/node_modules/rxjs": {
+      "version": "6.6.7",
+      "resolved": "https://registry.npmjs.org/rxjs/-/rxjs-6.6.7.tgz",
+      "integrity": "sha512-hTdwr+7yYNIT5n4AMYp85KA6yw2Va0FLa3Rguvbpa4W3I5xynaBZo41cM3XM+4Q6fRMj3sBYIR1VAmZMXYJvRQ==",
+      "dev": true,
+      "license": "Apache-2.0",
+      "dependencies": {
+        "tslib": "^1.9.0"
+      },
+      "engines": {
+        "npm": ">=2.0.0"
+      }
+    },
```
Lets revert the package-lock.json changes.

Brakebein commented:
> > It needs to be a throttled render function instead of debounced one, otherwise we won't properly see any updates.
> 
> This is indicative of bug - specifically #1081. Debouncing should work just fine so I've adjusted the demo to use it and I'll review this PR once I've fixed the #1081 issue.

Okay, this is a more elegant approach using requestAnimationFrame(). This ensures that the maximum FPS is used. The workaround I had in the earlier approach somehow solved the initial loading problem, which is most certainly due to the issue you want to solve in #1081.

Regarding _debounce_ vs _throttle_ (which is probably only a naming issue): from my understanding, your solution also aims to throttle the render calls, i.e., to reduce the number of calls to the maximum number of FPS.
- When _throttle_ is being called it fires your function periodically (eg. every 20ms).
- _Debounce_ only fires at the end (eg. 20ms after the last event).

[Difference between throttling and debouncing a function](https://stackoverflow.com/questions/25991367/difference-between-throttling-and-debouncing-a-function)

gkjohnson commented:
I just fixed #1081 and adjusted this PR with the new event needed so it should be good now. The new event will be available in the new release.

And you're right, I'll also change the function name to "throttle" since it's not resetting a timer every call.
